src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests_WithDigitSeparators.cs

```diff
@@ -0,0 +1,2137 @@
+        Console.WriteLine(d_29_6)
+    End Sub
+End Module
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [Fact]
+        [WorkItem(5529, "DevDiv_Projects/Roslyn")]
+        [Trait(Traits.Feature, Traits.Features.ReduceTokens)]
+        [Trait(Traits.Feature, "WithDigitSeparators")]
+        public async Task ReduceDecimalLiterals_30Digits()
+        {
+            var code = @"[|
+Module Program
+    Sub Main(args As String())
+        ' CATEGORY 2: 30 significant digits
+        ' Dev11 & Roslyn have identical behavior: pretty listed and round off to <= 29 significant digits
+        
+        Const d_30_1 As Decimal = .123_456_789_012_345_678_901_234_567_891D         ' Dev11 & Roslyn: 0.1234567890123456789012345679D
+        Const d_30_2 As Decimal = 0.123_456_789_012_345_678_901_234_568_789_1D      ' Dev11 & Roslyn: 0.1234567890123456789012345688D
+        Const d_30_3 As Decimal = 1.234_567_890_123_456_789_012_345_678_91D         ' Dev11 & Roslyn: 1.2345678901234567890123456789D
+        Const d_30_4 As Decimal = 123_456_789_012_345.678_901_234_567_891D          ' Dev11 & Roslyn: 123456789012345.67890123456789D
+        Const d_30_5 As Decimal = 12_345_678_901_234_567_890_123_456_789.1D         ' Dev11 & Roslyn: 12345678901234567890123456789D
+
+        ' Overflow case 30 significant digits before decimal place: Ensure no pretty listing.
+        Const d_30_6 As Decimal = 123_456_789_012_345_678_901_234_567_891.0D        ' Dev11 & Roslyn: 123456789012345678901234567891.0D
+
+        Console.WriteLine(d_30_1)
+        Console.WriteLine(d_30_2)
+        Console.WriteLine(d_30_3)
+        Console.WriteLine(d_30_4)
+        Console.WriteLine(d_30_5)
+        Console.WriteLine(d_30_6)
+    End Sub
+End Module
+|]";
+
+            var expected = @"
+Module Program
+    Sub Main(args As String())
+        ' CATEGORY 2: 30 significant digits
+        ' Dev11 & Roslyn have identical behavior: pretty listed and round off to <= 29 significant digits
+
+        Const d_30_1 As Decimal = .123_456_789_012_345_678_901_234_567_891D         ' Dev11 & Roslyn: 0.1234567890123456789012345679D
+        Const d_30_2 As Decimal = 0.123_456_789_012_345_678_901_234_568_789_1D      ' Dev11 & Roslyn: 0.1234567890123456789012345688D
+        Const d_30_3 As Decimal = 1.234_567_890_123_456_789_012_345_678_91D         ' Dev11 & Roslyn: 1.2345678901234567890123456789D
+        Const d_30_4 As Decimal = 123_456_789_012_345.678_901_234_567_891D          ' Dev11 & Roslyn: 123456789012345.67890123456789D
+        Const d_30_5 As Decimal = 12_345_678_901_234_567_890_123_456_789.1D         ' Dev11 & Roslyn: 12345678901234567890123456789D
+
+        ' Overflow case 30 significant digits before decimal place: Ensure no pretty listing.
+        Const d_30_6 As Decimal = 123_456_789_012_345_678_901_234_567_891.0D        ' Dev11 & Roslyn: 123456789012345678901234567891.0D
+
+        Console.WriteLine(d_30_1)
+        Console.WriteLine(d_30_2)
+      
```

Comment:
we already have TestWorkspace for use in tests.  Is there a reason not to use that here?